pkg/validations/upgradevalidations/immutableFields.go

```diff
@@ -43,8 +43,20 @@ func ValidateImmutableFields(ctx context.Context, k validations.KubectlClient, c
 		if err != nil {
 			return err
 		}
-		if !prevGitOps.Spec.Equal(&spec.GitOpsConfig.Spec) {
```

Comment:
For my own understanding, if we are automating the changes for the user, why do we need this change? They shouldn't need to update the cluster config file manually, right?

Reply:
they will need to if they don't use default clusterConfigPath